### DISCUSSION TITLE
Multiple dates on CalendarDatePicker

### DISCUSSION THREAD

**Discussion by shaidas** [AvaloniaUI/Avalonia#6595] [Q&A]
I need to highlight multiple days in the calendar to visually indicate existing appointments. I noticed there is a CalendarSelectionMode Enum but cannot find a way to highlight multiple dates. Anyone has any ideas if this is possible at all ?

**Comment by maxkatz6** [comment]
I don't see any ready API for that, but you can subscribe on DataContextChanged event from CalendarDayButton class. Before that you will need to get buttons list of this type traversing through visual tree probably. 
And in the handler set style-class depending on data context value, which you can use in your styles.

Can't provide an example now, but shouldn't be too complex.

**Comment by shaidas** [comment]
Thanks for taking the time out to reply. From your answer it looks like WPF and AvalonUI share similar classes and events when it comes to naming. We had implemented this in WPF using data triggers and CalendarDayButton class. I think the behaviours implementation may allow us to do the same thing in Avalon UI. I shall post back with some code to help others if we get a working solution.

**Comment by shaidas** [comment]
Have been trying it out. I can see the visual tree in the debugger. The CalendarDayButton are in a grid called MonthView. I have hooked on to the CalendarOpened event of the CalendarDatePicker and used the GetVisualDescendants method . I get 4 decscendant grids which seems to match the Visual tree. That is where the good news ends. Bizzarely all grids have null names and none have a CalendarDayButton descendant. Not sure if the event is incorrect or the getvisualdescendants is getting children and not descendants

**Comment by shaidas** [comment]
Also tried hooking on a Mutibinding converter to try and set the background style from XAML but just cannot get the date value to come through, no matter what source/relativesource and Path combo I try.  Well if anyone has any ideas let me know.
